### PR TITLE
SVCPLAN-4464: BREAKING - transition dependencies from storage-specifi…

### PR DESCRIPTION
**Pull request by jakerundall**
…c to general

The following parameters have been moved/renamed. Please adjust your environment Hiera accordingly:
- profile_slurm::compute::storage::require_storage -> profile_slurm::compute::require_dependencies
- profile_slurm::compute::storage::storage_dependencies -> profile_slurm::compute::dependencies
- profile_slurm::scheduler::storage::require_storage -> profile_slurm::scheduler::require_dependencies
- profile_slurm::scheduler::storage::storage_dependencies -> profile_slurm::scheduler::dependencies

### CODE REVIEW

**Comment by jakerundall** [comment]
Tested on Magnus on mgsched2 (test scheduler), mgtest02 (test compute), mg093 (GPU compute), mgtesthn1 (test HN, not affected).

File: manifests/compute.pp [hunk 1]
```diff
@@ -1,9 +1,37 @@
...
+#     "Gpfs::Bindmount['/sw']"
+#     "Gpfs::Bindmount['/u']"
+#     "Gpfs::Nativemount['/cluster']"
+#     "Lvm::Logical_volume[local]"
+#     "Profile_lustre::Nativemount_resource['/projects']"
+#     "Profile_lustre::Nativemount_resource['/taiga']"
+#
+# @param require_dependencies
+#   Optionally list resources (e.g., services) that require any of the
+#   items from the $dependencies parameter in order to function.
```
Comment (bsper2) [inline manifests/compute.pp:18]: The name of `require_dependencies` is a little confusing. If I was just glancing though hiera the first thing that would come to mind when seeing that param would be "list of dependencies required to make the module work". Then if I saw the regular dependencies in `profile_slurm::compute::dependencies:` I'd get pretty confused.

Granted you do a good job explaining the difference between the dependencies in the code comments, but if the param names were changed so you could figure out the differences just based of their names that would be a big bonus.

Maybe we keep `dependencies` and then rename the other one to `require_after_dependencies`. Which would make it clear that we have our normal dependencies list, and then a second layer of dependencies that should happen after the normal ones.

**Review by bsper2 — CHANGES_REQUESTED** [review]

File: manifests/compute.pp (hunk 1 above)
Comment (jakerundall) [reply manifests/compute.pp]: Per discussion in Slack I changed `require_dependencies` to `configure_after_dependencies`. I retested and it seems all good.

**Comment by jakerundall** [comment]
Ready for another look. Once ready to merge I'll rebase and squash the commits.

**Review by bsper2 — APPROVED** [review]
Looks good thanks.

**Pull request merged**